rekallpy/test/test_interval_set.py

```diff
@@ -466,6 +466,28 @@ def test_coalesce(self):
         self.assertIntervalSetEq(is1.coalesce(('t1', 't2'), Bounds3D.span,
             payload_plus,epsilon=2),
             IntervalSet([Interval(Bounds3D(1,23),payload=8)]))
+    
+    def test_coalesce_with_pred(self):
+        def overlapping_bboxes(intrvl1, intrvl2):
+            if Bounds3D.X(overlaps())(intrvl1, intrvl2) and Bounds3D.Y(overlaps())(intrvl1, intrvl2):
+                return True
+            else:
+                return False
+        is1 = IntervalSet([
+            Interval(Bounds3D(1,10,0.3,0.4,0.3,0.6),1),
+            Interval(Bounds3D(2,5,0.2,0.8,0.2,0.4),1),
+            Interval(Bounds3D(10,11,0.16,0.17,0.3,0.5),1),
```

Comment:
I would make this interval start at time 9 just to make it extra clear that it overlaps with one of the previous ones, but purposely isn't being merged in.